### PR TITLE
Specific the pod cluster env for xgboost rabit settup

### PR DESCRIPTION
**Pull request by merlintang**
## what is proposed in this PR ?
setup the cluster env for the pod, and export this information for xgboost rabit settup. 

## how to test this PR
provide the pod_test.go, and test passed.

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/xgboost-operator/23)

### CODE REVIEW

**Comment by merlintang** [comment]
@terrytangyuan , can you review this as well when you are free.

File: pkg/controller/xgboostjob/pod.go [hunk 1]
```diff
@@ -72,3 +74,55 @@ func convertPodList(list []corev1.Pod) []*corev1.Pod {
...
+	masterAddr := computeMasterAddr(xgboostjob.Name, strings.ToLower(string(v1alpha1.XGBoostReplicaTypeMaster)), strconv.Itoa(0))
+
+	masterPort, err := GetPortFromXGBoostJob(xgboostjob, v1alpha1.XGBoostReplicaTypeMaster)
+	if err != nil {
+		return err
+	}
+
+	totalReplicas := computeTotalReplicas(xgboostjob)
+
+	println(totalReplicas)
```
Comment (terrytangyuan) [inline pkg/controller/xgboostjob/pod.go]: Remove debug print statements

File: pkg/controller/xgboostjob/util.go [hunk 2]
```diff
@@ -44,6 +49,44 @@ func getDelegatingReader(c client.Client) (*client.DelegatingReader, error) {
 	return dr, nil
 }
 
+func computeMasterAddr(jobName, rtype, index string) string {
+	n := jobName + "-" + rtype + "-" + index
+	return strings.Replace(n, "/", "-", -1)
+}
+
+// GetPortFromPyTorchJob gets the port of pytorch container.
```
Comment (terrytangyuan) [inline pkg/controller/xgboostjob/util.go]: pytorch -> xgboost

File: pkg/controller/xgboostjob/pod_test.go [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+	return job
+}
+
+func NewXGBoostReplicaSpecTemplate() v1.PodTemplateSpec {
+	return v1.PodTemplateSpec{
+		Spec: v1.PodSpec{
+			Containers: []v1.Container{
+				v1.Container{
+					Name:  v1alpha1.DefaultContainerName,
+					Image: "test-image-for-kubeflow-pytorch-operator:latest",
```
Comment (terrytangyuan) [inline pkg/controller/xgboostjob/pod_test.go]: pytorch -> xgboost

**Comment by merlintang** [comment]
@terrytangyuan typro is fixed.

**Pull request merged**